crates/oro-tree/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+use memmap::MmapOptions;
+use semver::{Version, VersionReq};
+use serde::{
+    de::{Deserializer, Error as SerdeError},
+    Deserialize,
+};
+use ssri::Integrity;
+use std::fs::File;
+use std::{collections::HashMap, path::Path};
+use thiserror::Error;
+
+fn default_as_false() -> bool {
+    false
+}
+
+fn parse_integrity<'de, D>(deserializer: D) -> Result<Integrity, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+    s.parse().map_err(D::Error::custom)
+}
+
+fn parse_version<'de, D>(deserializer: D) -> Result<Version, D::Error>
```

Comment:
Likewise, `Version` already implements `Deserialize` so you don't really need this.